### PR TITLE
Fix demo

### PR DESCRIPTION
**Pull request by kyessenov**
Please take over this work. I'm not good at bash and have strong aversion to it.

---
This change is [Reviewable](https://reviewable.io/reviews/istio/istio/112)

### CODE REVIEW

File: scripts/update.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+# set env variables from the version file
+source $VERSION_FILE
+
+pushd $ROOT/kubernetes/istio-install
+for f in templates/*; do
+  cp -f $f .
+  sed -i "s|\$MANAGER_HUB|${MANAGER_HUB}|g" $(basename $f)
+  sed -i "s|\$MANAGER_TAG|${MANAGER_TAG}|g" $(basename $f)
+  sed -i "s|\$MIXER|${MIXER}|g" $(basename $f)
```
Comment (frankbu) [inline scripts/update.sh]: Do we really need to maintain a separate set of templates for this simple change? Why not simply update the actual files directly? No need for any templates.
```
sed -i -s "s|image: .*/|image: $MANAGER_HUB/|;s|image: \(.*\):.*|image: \1:$MANAGER_TAG|" istio-manager.yaml
sed -i -s "s|image: .*/|image: $MANAGER_HUB/|;s|image: \(.*\):.*|image: \1:$MANAGER_TAG|" istio-ingress-controller.yaml
sed -i "s|image: .*|image: $MIXER|" istio-mixer.yaml
```

**Comment by frankbu** [comment]
This PR overlaps with https://github.com/istio/istio/pull/107 but is missing many of the fixes in it, and therefore isn't close to actually working. I like the new format of istio.VERSION, but not sure how to get all the other fixes from #107 into here. How about we close this one, and I'll take the improvements and merge them into #107?

File: scripts/update.sh (hunk 1 above)
Comment (kyessenov) [inline scripts/update.sh]: The problem was that MIXER_HUB and MANAGER_HUB are the same, and cannot be replaced with two values.

**Comment by kyessenov** [comment]
Sure, feel free to take the changes here and incorporate them in #107. 
I also recommend removing `istioctl` binary and just fetching it before the test.

File: tests/istio/istio-ingress-controller.diff [hunk 2]
```diff
@@ -1,9 +0,0 @@
-35,36d35
```
Comment (sebastienvas) [inline tests/istio/istio-ingress-controller.diff]: Why don't we need this anymore ?

File: tests/istioUtils.sh [hunk 3]
```diff
@@ -18,8 +18,6 @@ ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)"
 TESTS_DIR="${ROOT}/tests"
 . ${TESTS_DIR}/commonUtils.sh || { echo "Cannot load common utilities"; exit 1; }
 
-ISTIOCLI="${ROOT}/demos/istioctl-linux -c ${HOME}/.kube/config"
```
Comment (sebastienvas) [inline tests/istioUtils.sh]: This was required when running in Jenkins.

File: tests/kubeUtils.sh [hunk 4]
```diff
@@ -35,9 +35,13 @@ function create_namespace(){
 
 # Bring up control plane
 function deploy_istio() {
-    local istio_install="${1}"
     print_block_echo "Deploying ISTIO"
-    ${K8CLI} -n ${NAMESPACE} create -f "${istio_install}" \
+    # order is important
+    ${K8CLI} -n ${NAMESPACE} create -f $ROOT/kubernetes/istio-install/istio-manager.yaml \
```
Comment (sebastienvas) [inline tests/kubeUtils.sh]: Why not using one command for all. If you want to keep that way, then change "Failed to create control plane".

**Comment by sebastienvas** [comment]
The test is failing because bookinfo.yaml cannot be found in dir. Something wrong with generate_bookinfo_yaml

**Comment by frankbu** [comment]
These changes moved into #107

**Comment by wattli** [comment]
@wlu2016

**Pull request closed without merging**